services/api/src/resources/notification/resolvers.ts

```diff
@@ -711,9 +711,55 @@ export const getAllNotifications: ResolverFn = async (
   args,
   { sqlClientPool, hasPermission }
 ) => {
-  await hasPermission('notification', 'viewAll');
+  let rows;
+
+  if (args.name && args.type) {
+    rows = await query(sqlClientPool, Sql.selectNotificationByNameAndType(args.name, args.type));
+
+    if (rows.length === 0) {
+      throw new Error(`No notification found for ${args.name} & ${args.type}`);
```

Comment:
To prevent some issues with enumeration etc, it would be better to throw a generic unauthorized error like this
```
throw new Error(`Unauthorized: You don't have permission to "view" on "notification"`);
```